### ISSUE TITLE
updates and upgrades should be done with existing profiles

### ISSUE THREAD

**Issue by Shitennouji** [MrAlex94/Waterfox#2367]
> You should be fine deleting that folder, although it may get recreated again by extensions and/or update caches.

Thanks MrAlex94 for the reply. 
However, This thread was closed without the consent of the Issue Submitter along with a simple answer, but I am not satisfied with that answer.

I'm aware that a new installation will create a new profile, but this is the first time I've experienced a case where an update or upgrade “replaces” a folder with a new profile or a new folder in a different path.
I'm a long-time regular user of Firefox, and use LibreWolf and Pale Moon, and I don't know of any such cases!

From your answer, is this a normal thing in Waterfox?
If updates and upgrades do not update existing profiles, then “personal data and customizations from before, such as settings, bookmarks, extensions, etc.” will not be inherited, and it is very inconvenient to have to migrate, reconfigure, and re-add them.

My original topic was that I saw this case as an accidental and unexpected incident (inadequate quality control from G3 to G4), but if it will continue to occur normally, then the “topic” will change. I will rephrase it as, I would like to see these low-tech methods of behavior changed.
In short, the argument is that “updates and upgrades should be done with existing profiles”.

Related Articles:
Dedicated profiles per Firefox installation | Firefox Help
https://support.mozilla.org/en-US/kb/dedicated-profiles-firefox-installation#firefox:win10:fx91
Back up and restore information in Firefox profiles | Firefox Help
https://support.mozilla.org/en-US/kb/back-and-restore-information-firefox-profiles

_Originally posted by @Shitennouji in https://github.com/WaterfoxCo/Waterfox/issues/2359#issuecomment-951596004_

**Comment by MrAlex94** [comment]
The reason I closed the previous issue is because I want the issue tracker to be as focused on specific issues as possible, rather than general discussion. I will close this as well since it is just discussion as well. Most projects do the same. 

>  If updates and upgrades do not update existing profiles, then “personal data and customizations from before, such as settings, bookmarks, extensions, etc.” will not be inherited, and it is very inconvenient to have to migrate, reconfigure, and re-add them.

It was a mistake, of course which is why it was rectified in a matter of hours. 

>  In short, the argument is that “updates and upgrades should be done with existing profiles”.

There's no argument, that is exactly what is and should be happening.

Essentially what had happened was that G4 was using the [MOZ_APP_VENDOR](https://github.com/WaterfoxCo/Waterfox/blob/current/browser/confvars.sh#L6) name because [MOZ_APP_PROFILE](https://github.com/WaterfoxCo/Waterfox/blob/current/browser/branding/waterfox/configure.sh#L6) was missing. We missed this during testing because the Dokimi branding had this define set [correctly](https://github.com/WaterfoxCo/Waterfox/blob/current/browser/branding/dokimi/configure.sh#L6).

**Comment by Shitennouji** [comment]
This answer is satisfactory.
I also understand “Closed”.
Thank you very much.
